### PR TITLE
[ZEPPELIN-6073][BUILD] Canonicalize binary distribution tarball name

### PR DESCRIPTION
**Pull request by pan3793**
### What is this PR for?

This PR aims to canonicalize the binary distribution tarball name assembled by Maven.

Currently, the `./mvnw clean package -Pbuild-distr -DskipTests` produces `zeppelin-0.12.0-SNAPSHOT.tar.gz`, while the final release tarballs are `zeppelin-0.11.1-bin-all.tgz` and `zeppelin-0.11.1-bin-netinst.tgz`.

After this change, `./mvnw clean package -Pbuild-distr -DskipTests` produces `zeppelin-0.12.0-SNAPSHOT-bin.tgz`.

### What type of PR is it?

Improvement

### What is the Jira issue?

ZEPPELIN-6073

### How should this be tested?

Pass existing GHA and manually run maven command to package the binary tarball.

```
$ ./mvnw clean package -Pbuild-distr -DskipTests
```

```
...
[INFO] -------------< org.apache.zeppelin:zeppelin-distribution >--------------
[INFO] Building Zeppelin: Packaging distribution 0.12.0-SNAPSHOT        [64/64]
[INFO] --------------------------------[ pom ]---------------------------------
...
[INFO] --- maven-assembly-plugin:3.7.1:single (make-assembly) @ zeppelin-distribution ---
[INFO] Reading assembly descriptor: src/assemble/distribution.xml
[INFO] Copying files to /Users/chengpan/Projects/apache-zeppelin/zeppelin-distribution/target/zeppelin-0.12.0-SNAPSHOT-bin
[INFO] 838 files copied to /Users/chengpan/Projects/apache-zeppelin/zeppelin-distribution/target/zeppelin-0.12.0-SNAPSHOT-bin
[INFO] Building tar: /Users/chengpan/Projects/apache-zeppelin/zeppelin-distribution/target/zeppelin-0.12.0-SNAPSHOT-bin.tgz
...
```

### Questions:
* Does the license files need to update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -162,7 +162,7 @@
 
     <!-- plugin versions -->
     <plugin.antrun.version>1.8</plugin.antrun.version>
-    <plugin.assembly.version>3.2.0</plugin.assembly.version>
+    <plugin.assembly.version>3.7.1</plugin.assembly.version>
```
Comment (pan3793) [inline pom.xml:165]: upgrade maven assembly plugin to support "tgz" suffix

File: zeppelin-distribution/src/assemble/distribution.xml [hunk 2]
```diff
@@ -57,7 +55,6 @@
         <include>README.md</include>
         <include>LICENSE*</include>
         <include>NOTICE</include>
-        <include>DISCLAIMER</include>
```
Comment (pan3793) [inline zeppelin-distribution/src/assemble/distribution.xml:60]: this file is required during the project incubating stage, not required after graduating as TLP(top-level project)

**Comment by Reamer** [comment]
Do you know why there are two versions: `netinst` and `bin-all`? I don't see any change that affects `netinst`.

**Comment by pan3793** [comment]
@Reamer the release packaging process is defined at `dev/create_rekease.sh`'s `function make_binary_release()`

`zeppelin-<version>-bin-netinst.tgz` and `zeppelin-<version>-bin-all.tgz` use different profiles to ship different set of interpreters.

```
make_binary_release netinst "-Pweb-classic -pl !hbase,!jdbc,!file,!flink,!cassandra,!elasticsearch,!bigquery,!alluxio,!livy,!groovy,!java,!neo4j,!sparql,!mongodb,!shell -am"
```

```
make_binary_release all "-Pweb-classic -pl !shell"
```

the main logic of `make_binary_release` are: 1) call `./mvnw clean package -Pbuild-distr -DskipTests ${BUILD_FLAGS}` to produce `zeppelin-<version>.tar.gz`, 2) untar, add license files, and re-tar, but with a different name pattern.

I guess most users who use customized build of Zeppelin would just call mvn command to produce the `zeppelin-<version>.tar.gz`, and most normal users would just download the `zeppelin-<version>-bin-netinst.tgz` or `zeppelin-<version>-bin-all.tgz`. This PR reduces the gap between the tarball's file name between those two type of users. For me, it makes life easier to write a [unified script](https://github.com/awesome-kyuubi/hadoop-testing) to test both the master branch and RC versions.

**Review by Reamer — APPROVED** [review]
Okay, now I've got most of the changes figured out. I only use the Dockerfile myself :grin:

LGTM

**Pull request merged**